pyupgrade.py

```diff
@@ -1295,7 +1301,8 @@ def visit_ClassDef(self, node):  # type: (ast.ClassDef) -> None
         self._class_info_stack.pop()
 
     def _visit_func(self, node):
-        # type: (Union[ast.FunctionDef, ast.Lambda, AsyncFunctionDef]) -> None
+        # type: (SyncOrAsyncFunctionDef) -> None
+        self._current_func, orig = node, self._current_func
```

Comment:
instead you can probably set `_current_func` to `None` inside `visit_AsyncFunctionDef`

Reply:
Addressed in a070506d51ca8e715a83912e6adaa8f6fbcc0b85.